client/starwhale/core/dataset/tabular.py

```diff
@@ -177,48 +153,39 @@ def _do_validate(self) -> None:
         if self.id == "":
             raise FieldTypeOrValueError("id is empty")
 
-        if not isinstance(self.annotations, dict) or not self.annotations:
-            raise FieldTypeOrValueError("no annotations field")
-
-        # TODO: add annotation items type check
-
-        if not self.data_link:
-            raise FieldTypeOrValueError("no raw_data_link field")
-
-        if not isinstance(self.data_format, DataFormatType):
-            raise NoSupportError(f"data format: {self.data_format}")
+        if not isinstance(self.data, dict) or not self.data:
+            raise FieldTypeOrValueError("no data field")
 
         if not isinstance(self.data_origin, DataOriginType):
             raise NoSupportError(f"data origin: {self.data_origin}")
 
-        if not isinstance(self.object_store_type, ObjectStoreType):
-            raise NoSupportError(f"object store {self.object_store_type}")
-
     def __str__(self) -> str:
-        return f"row-{self.id}, data-{self.data_link}, origin-[{self.data_origin}]"
+        return f"row-{self.id}"
 
     def __repr__(self) -> str:
-        return (
-            f"row-{self.id}, data-{self.data_link}(offset:{self.data_offset}, size:{self.data_size},"
-            f"format:{self.data_format}, meta type:{self.data_type}), "
-            f"origin-[{self.data_origin}], object store-{self.object_store_type}"
-        )
+        return f"row-{self.id}" f"{self.data} "
 
     def asdict(self, ignore_keys: t.Optional[t.List[str]] = None) -> t.Dict:
-        d = super().asdict(
-            ignore_keys=ignore_keys
-            or ["annotations", "extra_kw", "data_type", "data_link"]
-        )
+        d = super().asdict(ignore_keys=ignore_keys or ["data", "extra_kw"])
         d.update(_do_asdict_convert(self.extra_kw))
-        for k, v in self.annotations.items():
-            d[f"{self.ANNOTATION_PREFIX}{k}"] = JsonDict.from_data(v)
-        # TODO: use data_store SwObject to store data_type
-        d["data_type"] = json.dumps(
-            _do_asdict_convert(self.data_type), separators=(",", ":")
-        )
-        d["data_link"] = self.data_link
+        for k, v in self.data.items():
+            d[f"{self.DATA_PREFIX}{k}"] = JsonDict.from_data(v)
         return d
 
+    @classmethod
+    def artifacts_of_data(cls, content: t.Dict) -> t.List[BaseArtifact]:
+        artifacts = []
+        for _, v in content.items():
```

Comment:
`for _, v in content.items()` --> `for v in content.values()`

Reply:
> `for _, v in content.items()` --> `for v in content.values()` ?

done